src/components/ui/date-field.tsx

```diff
@@ -0,0 +1,143 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+import dayjs from "@/Utils/dayjs";
+
+interface DateFieldProps {
+  date?: Date;
+  onChange?: (date?: Date) => void;
+  disabled?: boolean;
+  id: string;
+}
+
+export default function DateField({
+  date,
+  onChange,
+  disabled,
+  id,
+}: DateFieldProps) {
+  const { t } = useTranslation();
+
+  const [day, setDay] = useState(
+    date ? date.getDate().toString().padStart(2, "0") : "",
+  );
+  const [month, setMonth] = useState(
+    date ? (date.getMonth() + 1).toString().padStart(2, "0") : "",
+  );
+  const [year, setYear] = useState(date ? date.getFullYear().toString() : "");
```

Comment:
what happens when the parent component changes the `date` value? this wouldn't get updated right?